src/database/models/users.model.ts

```diff
@@ -0,0 +1,38 @@
+export interface User {
+    user_id: string;
+    first_name: string;
+    last_name: string;
+    email: string;
+    password: string | null;
+    google_id: string | null;
+    facebook_id: string | null;
+    role: string;
+    is_activated: boolean;
+}
+
+export interface CreateUserSchema {
+    user_id: string;
+    first_name: string;
+    last_name: string;
+    email: string;
+    password?: string | null;
+    google_id?: string | null;
+    facebook_id?: string | null;
```

Comment:
for any of this field, this means that its type could be either `undefined | null | string`. Personally, I'd suggest to avoid giving too many input choices as it may be difficult to handle all permutations of input down the line. 

Either 
`password: string | null` 
OR
`password?: string` 

will work fine